readd scilab-5.5.2

efferre79 opened the pull request:
I have worked to put in a good shape the old version 5.5.2 that has been recently removed, I think the work is very close to have something installable (runtime tests are needed of course...). I know it's old but I think it currently requires less effort than latest 6.1.0. As soon as latest 6.1.0 is ready we can remove the old one!

Version 6.1.0

- it requires at least lucene-6.4.0 (my guess) which is not in portage so another package to take care. I have patched version 3.6.2 as described in https://bugs.gentoo.org/579086 but discovered that the API is too old and has been changed with the years (scilab requires lucene-queryparser with the class org.apache.lucene.queryparser.classic.QueryParser which is not available with lucene-3.6.2)
- other issues described in https://github.com/gentoo/sci/pull/1026

Version 5.5.2

- it needs batik-1.9 so jython has to be fixed (I have put there a patch, https://bugs.gentoo.org/758881)
- a patch is still missing to compile against >=lapack-3.6.0 (two symbols have been deprecated and removed http://netlib.org/lapack/lapack-3.6.0.html). A fortran expert can have a look by replacing the code referencing to the functions DGEQPF and ZGEQPF. It seems that a similar error affects also latest version (upstream is not very reactive, http://bugzilla.scilab.org/show_bug.cgi?id=14277). Here it  is the last error I see:

```
/usr/lib/gcc/x86_64-pc-linux-gnu/9.3.0/../../../../x86_64-pc-linux-gnu/bin/ld: /var/tmp/portage/sci-mathematics/scilab-5.5.2-r1/work/scilab-5.5.2/modules/.libs/libscilab-cli.so: undefined reference to dgeqpf_
/usr/lib/gcc/x86_64-pc-linux-gnu/9.3.0/../../../../x86_64-pc-linux-gnu/bin/ld: /var/tmp/portage/sci-mathematics/scilab-5.5.2-r1/work/scilab-5.5.2/modules/.libs/libscilab-cli.so: undefined reference to zgeqpf_
collect2: error: ld returned 1 exit status
make[1]: *** [Makefile:1156: scilab-bin] Error 1
make[1]: Leaving directory '/var/tmp/portage/sci-mathematics/scilab-5.5.2-r1/work/scilab-5.5.2'
make: *** [Makefile:1541: all-recursive] Error 1
 * ERROR: sci-mathematics/scilab-5.5.2-r1::science failed (compile phase):
 *   emake failed
 * 
 * If you need support, post the output of `emerge --info '=sci-mathematics/scilab-5.5.2-r1::science'`,
 * the complete build log and the output of `emerge -pqv '=sci-mathematics/scilab-5.5.2-r1::science'`.
 * The complete build log is located at '/var/log/portage/build/sci-mathematics/scilab-5.5.2-r1:20210110-190102.log.gz'.
 * For convenience, a symlink to the build log is located at '/var/tmp/portage/sci-mathematics/scilab-5.5.2-r1/temp/build.log.gz'.
 * The ebuild environment file is located at '/var/tmp/portage/sci-mathematics/scilab-5.5.2-r1/temp/environment'.
 * Working directory: '/var/tmp/portage/sci-mathematics/scilab-5.5.2-r1/work/scilab-5.5.2'
 * S: '/var/tmp/portage/sci-mathematics/scilab-5.5.2-r1/work/scilab-5.5.2'
!!! When you file a bug report, please include the following information:
GENTOO_VM=openjdk-8  CLASSPATH="" JAVA_HOME="/usr/lib64/openjdk-8"
JAVACFLAGS="-source 1.5 -target 1.5" COMPILER=""
and of course, the output of emerge --info =scilab-5.5.2
```

AndrewAmmerlaan commented:
Sure, we can keep the old version around, PR looks good, a couple of things though:

```
MissingLicenseFile: version 5.5.2-r1: no matching license file: CeCILL-2.1

```
 Upstream says:
```
Scilab is a free software, released under the terms of the GNU General Public License (GPL) v2.0. Scilab was originally licensed under the terms of the CeCILL v2.1 and continues to be available under such terms. For more information, see the COPYING file.
```
Does this also apply to the 5.5.2 version? Or should we re-add the CeCILL-2.1 license as well?

```
 file.size [fatal]             1
 (37 KiB) sci-mathematics/scilab/files/scilab-5.5.2-bug15107.patch
```
Could we split up this patch, or host it somehwhere else? Otherwise the repoman checks will break again (and I have been working really hard to make repoman happy about ::science)

efferre79 commented:
> 
> 
> Sure, we can keep the old version around, PR looks good, a couple of things though:
> 
> ```
> MissingLicenseFile: version 5.5.2-r1: no matching license file: CeCILL-2.1
> ```
> 
> Upstream says:
> 
> ```
> Scilab is a free software, released under the terms of the GNU General Public License (GPL) v2.0. Scilab was originally licensed under the terms of the CeCILL v2.1 and continues to be available under such terms. For more information, see the COPYING file.
> ```
> 
> Does this also apply to the 5.5.2 version? Or should we re-add the CeCILL-2.1 license as well?

I have copied the lastest version in the overlay and forgot to run repoman :-) I think the website refers to latest version, looking into README_Unix file in the source dir:

```
 0 - COPYRIGHT NOTICE
 ********************

 Scilab is a free software released under the terms of the CeCILL license.
 This license is compatibility with the GNU General Public License (GPL).
 See the included license: "COPYING" in English
                           "COPYING-FR" in French

 Scilab is also using some codes with other licenses. Please see
 SCI/modules/*/license.txt for details.
```
so I am not sure what to do. Moreover there are different licenses for the modules, which is the procedure in this case?

> ```
>  file.size [fatal]             1
>  (37 KiB) sci-mathematics/scilab/files/scilab-5.5.2-bug15107.patch
> ```
> 
> Could we split up this patch, or host it somehwhere else? Otherwise the repoman checks will break again (and I have been working really hard to make repoman happy about ::science)
> 
> (also, we might as well bump the EAPI version while we're at it)

The patch is huge because a Makefile.in is directly patched (there is no bootstrap mechanism and the sources don't have all the files shipped). So automake is missing some .depend file defined in modules/scicos/Makefile.modelica.am, it can be created only after configure phase when the Makefile is available. Is there a common place where to put this huge external files?

AndrewAmmerlaan commented:
> so I am not sure what to do.

Looks like we'll have to add the Cecill-2.1 license back. 

>  Moreover there are different licenses for the modules, which is the procedure in this case?

The `LICENSES` variable should contain the licenses of all things that are installed: https://devmanual.gentoo.org/general-concepts/licenses/index.html

>  Is there a common place where to put this huge external files?

Usually they would be hosted somewhere on Gentoo infrastructure, but I don't have access. So unless one of the devs is willing to make the patch available somewhere, we'd have to look for an alternative. One way to do it would be to fork the scilab repository and apply the patch there, GitHub has this nice feature where you can download a simple patch file
from a PR or commit using: `https://www.github.com/<owner>/<repo>/commit/<commit-id>.patch` which we could use.

Or we just split the patch up into 3 or 4 parts, which should also make repoman happy.

AndrewAmmerlaan commented:
Also, does this version require batik-1.8? Because that is no longer in ::gentoo, see: https://github.com/gentoo/sci/issues/879

Otherwise we might have more luck with a version between 5.5.2 and 6.1.0? Perhaps we can give 6.0.0 a shot?

AndrewAmmerlaan commented:
> Also, does this version require batik-1.8? Because that is no longer in ::gentoo, see: #879
> 
> Otherwise we might have more luck with a version between 5.5.2 and 6.1.0? Perhaps we can give 6.0.0 a shot?

Never mind, the 6.0.0 version segfaults apparently: https://github.com/gentoo/sci/issues/774

efferre79 commented:
> Also, does this version require batik-1.8? Because that is no longer in ::gentoo, see: #879

I have not found any issue in compiling with batik-1.9 but until I am able to run scilab I cannot be sure. I think the user in the issue is referring to the dependency in the old ebuild which was hardcoded to batik-1.8

efferre79 commented:
> 
> 
> > Also, does this version require batik-1.8? Because that is no longer in ::gentoo, see: #879
> > Otherwise we might have more luck with a version between 5.5.2 and 6.1.0? Perhaps we can give 6.0.0 a shot?
> 
> Never mind, the 6.0.0 version segfaults apparently: #774

If 6.0.0 requires > lucene:3.6 we are in the same situation as for 6.1.0, in that case it's better to push towards 6.1.0. Unfortunately I am not a java developer so I am not able to write the ebuild for latest lucene which is required for scilab > 5.5.2

efferre79 commented:
The installation of scilab-5.5.2 requires batik which requires jython which needs a patch not in portage yet, should we introduce the fixed version of jython in the sci overlay too?

efferre79 commented:
I have played a little bit with the licenses, it's really a mess :-( I have written a stupid and unoptimzed (full of copy&paste) bash script to discover the licenses and this is the situation right now for scilab-5.5.2:

```
modules/demo_tools/license.txt - demo_tools (1)
  - 5 - CeCILL
modules/gui/license.txt - gui (3)
  - 5 - CeCILL
  - 20 - Apple-like
  - 67 - other
modules/fftw/license.txt - fftw (2)
  - 5 - CeCILL
  - 22 - other
modules/differential_equations/license.txt - differential_equations (2)
  - 7 - CeCILL
  - 85 - public-domain
modules/ui_data/license.txt - ui_data (1)
  - 4 - CeCILL
modules/external_objects/license.txt - external_objects (1)
  - 4 - CeCILL
modules/interpolation/license.txt - interpolation (2)
  - 6 - CeCILL
  - 21 - public-domain
modules/genetic_algorithms/license.txt - genetic_algorithms (1)
  - 4 - CeCILL
modules/jvm/license.txt - jvm (1)
  - 5 - CeCILL
modules/scinotes/license.txt - scinotes (1)
  - 4 - CeCILL
modules/signal_processing/license.txt - signal_processing (1)
  - 7 - CeCILL
modules/randlib/license.txt - randlib (4)
  - 7 - CeCILL
  - 23 - artistic-like
  - 35 - other
  - 59 - public-domain
modules/atoms/license.txt - atoms (1)
  -  - other
modules/sound/license.txt - sound (1)
  - 6 - CeCILL
modules/api_scilab/license.txt - api_scilab (1)
  - 4 - CeCILL
modules/special_functions/license.txt - special_functions (1)
  - 6 - CeCILL
modules/polynomials/license.txt - polynomials (2)
  - 5 - CeCILL
  - 21 - public-domain
modules/output_stream/license.txt - output_stream (1)
  - 5 - CeCILL
modules/string/license.txt - string (2)
  - 5 - CeCILL
  - 21 - other
modules/xcos/license.txt - xcos (1)
  - 4 - CeCILL
modules/modules_manager/license.txt - modules_manager (1)
  - 5 - CeCILL
modules/mexlib/license.txt - mexlib (1)
  - 6 - CeCILL
modules/spreadsheet/license.txt - spreadsheet (3)
  - 5 - CeCILL
  - 21 - other
  - 76 - GPL-like
modules/parameters/license.txt - parameters (1)
  - 4 - CeCILL
modules/completion/license.txt - completion (1)
  - 5 - CeCILL
modules/m2sci/license.txt - m2sci (1)
  - 5 - CeCILL
modules/data_structures/license.txt - data_structures (2)
  - 6 - CeCILL
  - 21 - CeCILL
modules/core/license.txt - core (5)
  - 6 - CeCILL
  - 21 - LGPL-like
  - 31 - other
  - 53 - LGPL-like
  - 97 - BSD-like
modules/symbolic/license.txt - symbolic (1)
  - 5 - CeCILL
modules/linear_algebra/license.txt - linear_algebra (3)
  - 5 - CeCILL
  - 24 - GPL-like
  - 51 - other
modules/preferences/license.txt - preferences (1)
  - 4 - CeCILL
modules/simulated_annealing/license.txt - simulated_annealing (1)
  - 4 - CeCILL
modules/statistics/license.txt - statistics (1)
  - 8 - CeCILL
modules/graph/license.txt - graph (1)
  - 4 - CeCILL
modules/windows_tools/license.txt - windows_tools (1)
  - 5 - CeCILL
modules/time/license.txt - time (1)
  - 5 - CeCILL
modules/xml/license.txt - xml (1)
  - 4 - CeCILL
modules/intersci/license.txt - intersci (1)
  - 5 - CeCILL
modules/boolean/license.txt - boolean (1)
  - 6 - CeCILL
modules/sparse/license.txt - sparse (2)
  - 7 - CeCILL
  - 23 - BSD-like
modules/action_binding/license.txt - action_binding (1)
  - 6 - CeCILL
modules/optimization/license.txt - optimization (4)
  - 6 - CeCILL
  - 21 - other
  - 42 - other
  - 73 - GPL-like
modules/scicos_blocks/license.txt - scicos_blocks (1)
  - 9 - GPL-like
modules/development_tools/license.txt - development_tools (1)
  - 5 - CeCILL
  - Found license file modules/development_tools/src/perl/xml2test/scilab_code_license.txt
  - Found license file modules/development_tools/src/perl/xml2test/gateway_code_license.txt
modules/tclsci/license.txt - tclsci (3)
  - 6 - CeCILL
  - 22 - other
  - 33 - GPL-like
modules/tclsci/tcl/sciGUI/license.txt - sciGUI (1)
  -  - other
modules/renderer/license.txt - renderer (1)
  - 5 - CeCILL
modules/matio/license.txt - matio (1)
  - 5 - CeCILL
modules/commons/license.txt - commons (1)
  - 4 - CeCILL
modules/scicos/license.txt - scicos (2)
  - 9 - GPL-like
  - 24 - other
modules/types/license.txt - types (1)
  - 4 - CeCILL
modules/external_objects_java/license.txt - external_objects_java (1)
  - 4 - CeCILL
modules/elementary_functions/license.txt - elementary_functions (6)
  - 7 - CeCILL
  - 23 - public-domain
  - 58 - other
  - 86 - BSD-like
  - 113 - BSD-like
  - 130 - public-domain
modules/graphics/license.txt - graphics (1)
  - 7 - CeCILL
modules/hdf5/license.txt - hdf5 (1)
  - 4 - CeCILL
modules/call_scilab/license.txt - call_scilab (2)
  - 5 - CeCILL
  - 22 - public-domain
modules/graphic_objects/license.txt - graphic_objects (1)
  - 4 - CeCILL
modules/dynamic_link/license.txt - dynamic_link (2)
  - 6 - CeCILL
  - 23 - public-domain
modules/overloading/license.txt - overloading (1)
  - 7 - CeCILL
modules/graphic_export/license.txt - graphic_export (1)
  - 5 - CeCILL
modules/history_manager/license.txt - history_manager (1)
  - 5 - CeCILL
modules/io/license.txt - io (1)
  - 6 - CeCILL
modules/fileio/license.txt - fileio (1)
  - 6 - CeCILL
modules/javasci/license.txt - javasci (2)
  - 5 - CeCILL
  - 22 - public-domain
modules/functions/license.txt - functions (1)
  - 6 - CeCILL
modules/helptools/license.txt - helptools (2)
  - 6 - CeCILL
  - 19 - CeCILL
modules/cacsd/license.txt - cacsd (7)
  - 6 - CeCILL
  - 25 - GPL-like
  - 48 - public-domain
  - 83 - BSD-like
  - 102 - public-domain
  - 111 - public-domain
  - 120 - public-domain
modules/parallel/license.txt - parallel (1)
  - 4 - CeCILL
modules/console/license.txt - console (1)
  - 5 - CeCILL
modules/compatibility_functions/license.txt - compatibility_functions (1)
  - 5 - CeCILL
modules/double/license.txt - double (1)
  - 5 - CeCILL
modules/arnoldi/license.txt - arnoldi (1)
  - 6 - CeCILL
modules/umfpack/license.txt - umfpack (2)
  - 13 - CeCILL
  - 27 - other
  - Found license file modules/umfpack/TAUCS_license.txt
  - Found license file modules/umfpack/UMFPACK_license.txt
modules/history_browser/license.txt - history_browser (1)
  - 4 - CeCILL
modules/localization/license.txt - localization (1)
  - 5 - CeCILL
modules/integer/license.txt - integer (1)
  - 5 - CeCILL

```

with _other_ the script is not able to recognize a license, with _*-like_ it means "one-of" group of licenses with similar name. I am not sure I want to read all those files and compare with the files in /usr/portage/licenses ....

efferre79 commented:
> > Is there a common place where to put this huge external files?
> 
> Usually they would be hosted somewhere on Gentoo infrastructure, but I don't have access. So unless one of the devs is willing to make the patch available somewhere, we'd have to look for an alternative. One way to do it would be to fork the scilab repository and apply the patch there, GitHub has this nice feature where you can download a simple patch file
> from a PR or commit using: `https://www.github.com/<owner>/<repo>/commit/<commit-id>.patch` which we could use.
> 
> Or we just split the patch up into 3 or 4 parts, which should also make repoman happy.

another option is to add to the gentoo github account a gist that can be referenced in the ebuild with a direct link. For instance, I have added the patch to my account in [scilab-5.5.2-bug15107.patch](https://gist.github.com/efferre79/fe88f84a90728f6fd8bbd1d561e96bfc), then in the ebuild

`SRC_URI="https://gist.github.com/gentoo/fe88f84a90728f6fd8bbd1d561e96bfc -> scilab-5.5.2-bug15107.patch"`

I know that the sci overlay on github is only a mirror but we can use it as a storage too. Of course it's better to move the patch in the gist section of the gentoo account on github, and not keeping in mine, also to have an official link :-) 

I don't know how long the files are preserved there but in my gist section I have files which are two years old right now.

AndrewAmmerlaan commented:
> The installation of scilab-5.5.2 requires batik which requires jython which needs a patch not in portage yet, should we introduce the fixed version of jython in the sci overlay too?

I don't think we should overwrite packages in ::gentoo, we can technically. But personally I'm not fond of overlays that change ::gentoo packages, so I would like to avoid that if at all possible.




> > > Also, does this version require batik-1.8? Because that is no longer in ::gentoo, see: #879
> > > Otherwise we might have more luck with a version between 5.5.2 and 6.1.0? Perhaps we can give 6.0.0 a shot?
> > 
> > 
> > Never mind, the 6.0.0 version segfaults apparently: #774
> 
> If 6.0.0 requires > lucene:3.6 we are in the same situation as for 6.1.0, in that case it's better to push towards 6.1.0. Unfortunately I am not a java developer so I am not able to write the ebuild for latest lucene which is required for scilab > 5.5.2

I tried 6.0.2 and got the exact same compile error as I got with 6.1.0:

```
/usr/include/unicode/ucnv.h:588:1: error: conflicting declaration of C function 'void icu::swap(icu::LocalUConverterPointer&, icu::LocalUConverterPointer&)'
588 | U_DEFINE_LOCAL_OPEN_POINTER(LocalUConverterPointer, UConverter, ucnv_close);
| ^~~~~~~~~~~~~~~~~~~~~~~~~~~
/usr/include/unicode/uenum.h:69:1: note: previous declaration 'void icu::swap(icu::LocalUEnumerationPointer&, icu::LocalUEnumerationPointer&)'
69 | U_DEFINE_LOCAL_OPEN_POINTER(LocalUEnumerationPointer, UEnumeration, uenum_close);
| ^~~~~~~~~~~~~~~~~~~~~~~~~~~
```

> with _other_ the script is not able to recognize a license, with _*-like_ it means "one-of" group of licenses with similar name. I am not sure I want to read all those files and compare with the files in /usr/portage/licenses ....

Me neither, lets just worry about getting this functional first, and then if and when it works we can worry about getting LICENSES correct

> another option is to add to the gentoo github account a gist that can be referenced in the ebuild with a direct link.

That would work too, but we'd need someone with access to the Gentoo GitHub account (my guess would be that the infra team has access, but I don't think we should bother them with this, they are busy enough as is). We did something similar in ::guru earlier: https://github.com/gentoo/guru/commit/3f826bf9b12a22fa0e873b7818193187f999b1b5 where we solved this by downloading the relevant patch from the corresponding PR, which works as long as no one changes that PR.

efferre79 commented:
> I tried 6.0.2 and 

Does 6.0.2 work with lucene:3.6?

> and got the exact same compile error as I got with 6.1.0:
> 
> ```
> /usr/include/unicode/ucnv.h:588:1: error: conflicting declaration of C function 'void icu::swap(icu::LocalUConverterPointer&, icu::LocalUConverterPointer&)'
> 588 | U_DEFINE_LOCAL_OPEN_POINTER(LocalUConverterPointer, UConverter, ucnv_close);
> | ^~~~~~~~~~~~~~~~~~~~~~~~~~~
> /usr/include/unicode/uenum.h:69:1: note: previous declaration 'void icu::swap(icu::LocalUEnumerationPointer&, icu::LocalUEnumerationPointer&)'
> 69 | U_DEFINE_LOCAL_OPEN_POINTER(LocalUEnumerationPointer, UEnumeration, uenum_close);
> | ^~~~~~~~~~~~~~~~~~~~~~~~~~~
> ```

That error should be related to icu-64 and C-linkage vs template problem. Try adapting the [patch](https://github.com/bartoszek/scilab/blob/master/libxml.patch) which is for 6.1.0

> > with _other_ the script is not able to recognize a license, with _*-like_ it means "one-of" group of licenses with similar name. I am not sure I want to read all those files and compare with the files in /usr/portage/licenses ....
> 
> Me neither, lets just worry about getting this functional first, and then if and when it works we can worry about getting LICENSES correct

What about merging this PR to give more visibility so that someone else can have a look? It will not be production ready but at least the effort can be shared. The ebuild is still without keywords so it's a good indication is not ready :-) 

We definitely need a fortran expert, I am not able to solve the issue due to deprecated lapack functions, the same problem is also valid for version 6.1.0 as described in the first post.

> > another option is to add to the gentoo github account a gist that can be referenced in the ebuild with a direct link.
> 
> That would work too, but we'd need someone with access to the Gentoo GitHub account (my guess would be that the infra team has access, but I don't think we should bother them with this, they are busy enough as is). We did something similar in ::guru earlier: [gentoo/guru@3f826bf](https://github.com/gentoo/guru/commit/3f826bf9b12a22fa0e873b7818193187f999b1b5) where we solved this by downloading the relevant patch from the corresponding PR, which works as long as no one changes that PR.

So basically there should be a PR whose only scope is adding a patch file to the repository, the PR will be kept alive and not merged forever. Did I understand correctly?

AndrewAmmerlaan commented:
> Does 6.0.2 work with lucene:3.6?

Probably not, the official distributed binary ships with version 5.2.1

> That error should be related to icu-64 and C-linkage vs template problem. Try adapting the [patch](https://github.com/bartoszek/scilab/blob/master/libxml.patch) which is for 6.1.0

Awesome, this patch fixes the issue. Sadly I now get a different error related to fortran (I don't know anything about fortran :cry: )

```
src/fortran/optml2.f:172:50:

162 |         if (info .gt. 1) call outl2(30,nq,nq,q,xx,t,tout)
|                                               2   
......
172 |         if (info .gt. 1) call outl2(31,nq,nbout,q,dnorm0,t,tout)
|                                                  1
Error: Rank mismatch between actual argument at (1) and actual argument at (2) (rank-1 and scalar)
```


> What about merging this PR to give more visibility so that someone else can have a look? It will not be production ready but at least the effort can be shared. The ebuild is still without keywords so it's a good indication is not ready :-)

I will do that, after we fix the issue with the patch, otherwise QA checks will be broken again for the entire repo.

> So basically there should be a PR whose only scope is adding a patch file to the repository, the PR will be kept alive and not merged forever. Did I understand correctly?

It can be merged and/or closed, the only restriction would be that nothing can be pushed to it. (If the patch changes emerge will fail to match the checksum in the manifest file). It doesn't even have to be a PR to the scihub repo, it can be a commit or a PR to a fork too.

AndrewAmmerlaan commented:
> Awesome, this patch fixes the issue. Sadly I now get a different error related to fortran (I don't know anything about fortran cry )

Fixed this one, now it almost compiles successfully now:
```
/usr/lib/gcc/x86_64-pc-linux-gnu/10.2.0/../../../../x86_64-pc-linux-gnu/bin/ld: /var/tmp/portage/sci-mathematics/scilab-6.0.2/work/scilab-6.0.2/modules/ast/.libs/libsciast.so: undefined reference to `types::Inspector::removeItem(types::InternalType*)'
/usr/lib/gcc/x86_64-pc-linux-gnu/10.2.0/../../../../x86_64-pc-linux-gnu/bin/ld: ./modules/.libs/libscilab-cli.so: undefined reference to `dgeqpf_'
/usr/lib/gcc/x86_64-pc-linux-gnu/10.2.0/../../../../x86_64-pc-linux-gnu/bin/ld: ./modules/.libs/libscilab-cli.so: undefined reference to `sci_inspectorShowUnreferencedItem(std::vector<types::InternalType*, std::allocator<types::InternalType*> >&, int, std::vector<types::InternalType*, std::allocator<types::InternalType*> >&)'
/usr/lib/gcc/x86_64-pc-linux-gnu/10.2.0/../../../../x86_64-pc-linux-gnu/bin/ld: ./modules/.libs/libscilab-cli.so: undefined reference to `sci_inspectorGetFunctionList(std::vector<types::InternalType*, std::allocator<types::InternalType*> >&, int, std::vector<types::InternalType*, std::allocator<types::InternalType*> >&)'
/usr/lib/gcc/x86_64-pc-linux-gnu/10.2.0/../../../../x86_64-pc-linux-gnu/bin/ld: ./modules/.libs/libscilab-cli.so: undefined reference to `sci_inspectorGetItemCount(std::vector<types::InternalType*, std::allocator<types::InternalType*> >&, int, std::vector<types::InternalType*, std::allocator<types::InternalType*> >&)'
/usr/lib/gcc/x86_64-pc-linux-gnu/10.2.0/../../../../x86_64-pc-linux-gnu/bin/ld: ./modules/.libs/libscilab-cli.so: undefined reference to `sci_inspectorGetUnreferencedItem(std::vector<types::InternalType*, std::allocator<types::InternalType*> >&, int, std::vector<types::InternalType*, std::allocator<types::InternalType*> >&)'
/usr/lib/gcc/x86_64-pc-linux-gnu/10.2.0/../../../../x86_64-pc-linux-gnu/bin/ld: ./modules/.libs/libscilab-cli.so: undefined reference to `sci_inspectorGetItem(std::vector<types::InternalType*, std::allocator<types::InternalType*> >&, int, std::vector<types::InternalType*, std::allocator<types::InternalType*> >&)'
/usr/lib/gcc/x86_64-pc-linux-gnu/10.2.0/../../../../x86_64-pc-linux-gnu/bin/ld: ./modules/.libs/libscilab-cli.so: undefined reference to `sci_inspectorDeleteUnreferencedItems(std::vector<types::InternalType*, std::allocator<types::InternalType*> >&, int, std::vector<types::InternalType*, std::allocator<types::InternalType*> >&)'
/usr/lib/gcc/x86_64-pc-linux-gnu/10.2.0/../../../../x86_64-pc-linux-gnu/bin/ld: ./modules/.libs/libscilab-cli.so: undefined reference to `sci_inspectorShowItem(std::vector<types::InternalType*, std::allocator<types::InternalType*> >&, int, std::vector<types::InternalType*, std::allocator<types::InternalType*> >&)'
/usr/lib/gcc/x86_64-pc-linux-gnu/10.2.0/../../../../x86_64-pc-linux-gnu/bin/ld: /var/tmp/portage/sci-mathematics/scilab-6.0.2/work/scilab-6.0.2/modules/ast/.libs/libsciast.so: undefined reference to `types::Inspector::addItem(types::InternalType*)'
collect2: error: ld returned 1 exit status
```

efferre79 commented:
> 
> 
> > Does 6.0.2 work with lucene:3.6?
> 
> Probably not, the official distributed binary ships with version 5.2.1

but at least it compiles so the API is not different, when I was trying compiling 6.1.0 I wasn't able to pass configure phase with lucene:3.6

> > What about merging this PR to give more visibility so that someone else can have a look? It will not be production ready but at least the effort can be shared. The ebuild is still without keywords so it's a good indication is not ready :-)
> 
> I will do that, after we fix the issue with the patch, otherwise QA checks will be broken again for the entire repo.

I should then create a commit containing only the patch so that can be referenced later, in my fork I can remove the patch and readd it later. Is this correct?

efferre79 commented:
> 
> 
> > Awesome, this patch fixes the issue. Sadly I now get a different error related to fortran (I don't know anything about fortran cry )
> 
> Fixed this one, now it almost compiles successfully:

But your are trying to compile 6.0.1 and adding patches to the ebuild for 6.1.0, is that the correct way to go?

>`/usr/lib/gcc/x86_64-pc-linux-gnu/10.2.0/../../../../x86_64-pc-linux-gnu/bin/ld: ./modules/.libs/libscilab-cli.so: undefined reference to dgeqpf_`

And this error is the same where I got stuck for 5.5.1 as foreseen :-)

AndrewAmmerlaan commented:
> > > Does 6.0.2 work with lucene:3.6?
> > 
> > 
> > Probably not, the official distributed binary ships with version 5.2.1
> 
> but at least it compiles so the API is not different, when I was trying compiling 6.1.0 I wasn't able to pass configure phase with lucene:3.6
> 
> > > What about merging this PR to give more visibility so that someone else can have a look? It will not be production ready but at least the effort can be shared. The ebuild is still without keywords so it's a good indication is not ready :-)
> > 
> > 
> > I will do that, after we fix the issue with the patch, otherwise QA checks will be broken again for the entire repo.
> 
> I should then create a commit containing only the patch so that can be referenced later, in my fork I can remove the patch and readd it later. Is this correct?

Yes, as long as the commit that applies the patch is available somewhere in the git history we can download it from github.



> But your are trying to compile 6.0.1 and adding patches to the ebuild for 6.1.0, is that the correct way to go?

I'm testing both 6.0.2 and 6.1.0 at the same time, the same patch applies to both versions without problems. (and is also needed in both versions).

efferre79 commented:
> 
> 
> > > > Does 6.0.2 work with lucene:3.6?
> > > 
> > > 
> > > Probably not, the official distributed binary ships with version 5.2.1
> > 
> > 
> > but at least it compiles so the API is not different, when I was trying compiling 6.1.0 I wasn't able to pass configure phase with lucene:3.6
> 
> 6.1.0 configures successfully with USE="-gui -xcos", I'm first testing things with all USE flags disabled, once we get that working we can try to get the features working that are enable with the USE flags.
> 
> > > > What about merging this PR to give more visibility so that someone else can have a look? It will not be production ready but at least the effort can be shared. The ebuild is still without keywords so it's a good indication is not ready :-)
> > > 
> > > 
> > > I will do that, after we fix the issue with the patch, otherwise QA checks will be broken again for the entire repo.
> > 
> > 
> > I should then create a commit containing only the patch so that can be referenced later, in my fork I can remove the patch and readd it later. Is this correct?
> 
> Yes, as long as the commit that applies the patch is available somewhere in the git history we can download it from github.
> 
> > But your are trying to compile 6.0.1 and adding patches to the ebuild for 6.1.0, is that the correct way to go?
> 
> I'm testing both 6.0.2 and 6.1.0 at the same time, the same patch applies to both versions without problems. (and is also needed in both versions).

out of curiosity, how can you pass the configure phase with the old lucene:3.6 in portage?

AndrewAmmerlaan commented:
> > I'm testing both 6.0.2 and 6.1.0 at the same time, the same patch applies to both versions without problems. (and is also needed in both versions).
> 
> out of curiosity, how can you pass the configure phase with the old lucene:3.6 in portage?

By setting USE="-gui -xcos"

AndrewAmmerlaan commented on sci-mathematics/scilab/scilab-5.5.2-r1.ebuild:
```diff
@@ -0,0 +1,298 @@
+
+DESCRIPTION="Scientific software package for numerical computations"
+HOMEPAGE="http://www.scilab.org/"
+SRC_URI="http://www.scilab.org/download/${PV}/${P}-src.tar.gz"
```
```suggestion
SRC_URI="https://www.scilab.org/download/${PV}/${P}-src.tar.gz
https://github.com/gentoo/sci/commit/4c2a07c4629c61395a998633ccfcb34d72569529.patch -> ${P}-bug15107.patch"
```

AndrewAmmerlaan replied:
Ooh no wait, that is the wrong URL

efferre79 replied:
I am doing

AndrewAmmerlaan replied:
This is the correct URL: https://raw.githubusercontent.com/efferre79/sci/4c2a07c4629c61395a998633ccfcb34d72569529/sci-mathematics/scilab/files/scilab-5.5.2-bug15107.patch

efferre79 replied:
and there is another problem, the patch from github creates a patch file in ${WORKDIR} that has to be applied in src_prepare(), that is an easy fix, the only drawback is that the user will see the message

`* Applying scilab-5.5.2-bug15107.patch ...`

AndrewAmmerlaan commented on sci-mathematics/scilab/scilab-5.5.2-r1.ebuild:
```diff
@@ -0,0 +1,298 @@
+# Copyright 1999-2020 Gentoo Authors
+# Distributed under the terms of the GNU General Public License v2
+
+EAPI=6
```
```suggestion
EAPI=7
```

AndrewAmmerlaan commented on sci-mathematics/scilab/scilab-5.5.2-r1.ebuild:
```diff
@@ -0,0 +1,298 @@
+JAVA_PKG_OPT_USE="gui"
+VIRTUALX_REQUIRED="manual"
+
+inherit autotools bash-completion-r1 check-reqs eutils flag-o-matic \
```
```suggestion
inherit autotools check-reqs eutils flag-o-matic \
```

There's not really any point in having this USE flag, I already removed it from the other ebuilds, because we shouldn't have USE flags toggle the installation of small files. There is no point in forcing an user to recompile the entire package just to add bash coimpletions

efferre79 replied:
not that line, but on the IUSE :-)

AndrewAmmerlaan replied:
ooh oops, my bad. It should indeed by IUSE of course, we do still need the eclass

AndrewAmmerlaan commented on sci-mathematics/scilab/scilab-5.5.2-r1.ebuild:
```diff
@@ -0,0 +1,298 @@
+	"${FILESDIR}/${P}-libxml-icu64.patch" # similar to https://bugzilla.scilab.org/show_bug.cgi?id=16588
+	"${FILESDIR}/${P}-hdf5-1.8.10.patch" # adapted from https://github.com/bartoszek/scilab
+	"${FILESDIR}/${P}-bug15449.patch" # http://bugzilla.scilab.org/show_bug.cgi?id=15449
+	"${FILESDIR}/${P}-ocaml-4.0.4.patch" # https://github.com/bartoszek/scilab
```
```suggestion
	"${FILESDIR}/${P}-ocaml-4.0.4.patch" # https://github.com/bartoszek/scilab
	"${DISTDIR}/${P}-bug15107.patch"
	
```

AndrewAmmerlaan commented on sci-mathematics/scilab/scilab-5.5.2-r1.ebuild:
```diff
@@ -0,0 +1,298 @@
+	default
+	prune_libtool_files --all
+	rm -rf "${D}"/usr/share/scilab/modules/*/tests ||die
+	use bash-completion && newbashcomp "${FILESDIR}"/"${PN}".bash_completion "${PN}"
```
```suggestion
	newbashcomp "${FILESDIR}"/"${PN}".bash_completion "${PN}"
```

AndrewAmmerlaan commented:
Perfect, this should do the trick, now all we need to do is re-add the missing license file:
```
  LICENSE.invalid [fatal]       1
   sci-mathematics/scilab/scilab-5.5.2-r1.ebuild: CeCILL-2.1
```

efferre79 commented:
after this is merged I will create an issue for the building failure due to lapack

efferre79 commented:
Please hold on, there is still an issue with USE=mpi active

efferre79 commented:
Ok, now is ready :-)

AndrewAmmerlaan commented:
> Ok, now is ready :-)

Looks good, I was going to merge this but then I ran into a small issue:

```
Resolving github.com... 140.82.121.4
Connecting to github.com|140.82.121.4|:443... connected.
HTTP request sent, awaiting response... 302 Found
Location: https://patch-diff.githubusercontent.com/raw/gentoo/sci/pull/1038.patch [following]
--2021-01-12 09:44:54--  https://patch-diff.githubusercontent.com/raw/gentoo/sci/pull/1038.patch
Resolving patch-diff.githubusercontent.com... 140.82.121.4
Connecting to patch-diff.githubusercontent.com|140.82.121.4|:443... connected.
HTTP request sent, awaiting response... 200 OK
Cookie coming from patch-diff.githubusercontent.com attempted to set domain to github.com
Length: unspecified [text/plain]
Saving to: ‘/tmp/tmp.wbNKZrntKF/all.patch’

/tmp/tmp.wbNKZrntKF/all.patch            [ <=>                                                                  ] 228.87K  --.-KB/s    in 0.09s

2021-01-12 09:44:54 (2.54 MB/s) - ‘/tmp/tmp.wbNKZrntKF/all.patch’ saved [234367]

Commit no. 0001 was not signed off by the author!
```

The Gentoo copyright policy requires that all commits are signed-off by their authors. This is because the author has to explicitly declare that his/her contributions can be modified /redistributed under the ebuilds licences (GPL-2): see https://www.gentoo.org/glep/glep-0076.html

You can easily sign-off commits by adding `SIGNED_OFF_BY=Your Name <your email-adres>` to `make.conf`, and using `repoman ci` instead of `git commit` for committing. You can also reword the commits and append the necessary sign-off line: https://docs.github.com/en/free-pro-team@latest/github/committing-changes-to-your-project/changing-a-commit-message

Also, the use of merge commits in Gentoo repositories is strongly discouraged: https://www.gentoo.org/glep/glep-0066.html#id6 and the implicit merge commits from `git pull` are explicitly forbidden. It would be best to use drop the merge commits and use rebase instead. See the wiki on how to configure git correctly: https://wiki.gentoo.org/wiki/Gentoo_git_workflow

efferre79 commented:
I think is better to reject this pull request, I'll redo another one with better git workflow

AndrewAmmerlaan commented:
> I think is better to reject this pull request, I'll redo another one with better git workflow

That works too :+1:

efferre79 commented:
> Resolving github.com... 140.82.121.4
> Connecting to github.com|140.82.121.4|:443... connected.
> HTTP request sent, awaiting response... 302 Found
> Location: https://patch-diff.githubusercontent.com/raw/gentoo/sci/pull/1038.patch [following]
> --2021-01-12 09:44:54--  https://patch-diff.githubusercontent.com/raw/gentoo/sci/pull/1038.patch
> Resolving patch-diff.githubusercontent.com... 140.82.121.4
> Connecting to patch-diff.githubusercontent.com|140.82.121.4|:443... connected.
> HTTP request sent, awaiting response... 200 OK
> Cookie coming from patch-diff.githubusercontent.com attempted to set domain to github.com
> Length: unspecified [text/plain]
> Saving to: ‘/tmp/tmp.wbNKZrntKF/all.patch’
> 
> /tmp/tmp.wbNKZrntKF/all.patch            [ <=>                                                                  ] 228.87K  --.-KB/s    in 0.09s
> 
> 2021-01-12 09:44:54 (2.54 MB/s) - ‘/tmp/tmp.wbNKZrntKF/all.patch’ saved [234367]
> 
> Commit no. 0001 was not signed off by the author!

Thanks for the git suggestions, but where is coming from 1038.patch?

AndrewAmmerlaan commented:
> Thanks for the git suggestions, but where is 1038.patch coming from ?

From this PR, there's this useful utility for merging Gentoo PRs called PRam: app-portage/pram It downloads the PR from here and applies it to my local version of ::science, except now it fails because the commits did not contain a sign-off line.

efferre79 commented:
> 
> 
> > Thanks for the git suggestions, but where is 1038.patch coming from ?
> 
> From this PR, there's this useful utility for merging Gentoo PRs called PRam: app-portage/pram It downloads the PR from here and applies it to my local version of ::science, except now it fails because the commits did not contain a sign-off line.

A grep in the folder doesn't find anything with that name, actually in the full sci overlay I have forked :-(

AndrewAmmerlaan commented:
> > > Thanks for the git suggestions, but where is 1038.patch coming from ?
> > 
> > 
> > From this PR, there's this useful utility for merging Gentoo PRs called PRam: app-portage/pram It downloads the PR from here and applies it to my local version of ::science, except now it fails because the commits did not contain a sign-off line.
> 
> A grep in the folder doesn't find anything with that name, actually in the full sci overlay I have forked :-(

It's not in the overlay, it is the PR itself, 1038 is the identifier number of this PR, 1038.patch is the patch that represents this PR. Anyway, that's not really relevant at the moment, the problem occurs because the changes *in* 1038.patch (aka this PR) are not signed-off. If you rebase and reword this to add the required sign-off lines it should work just fine.

efferre79 commented:
I have finally got to successfully install 5.5.2 with a few other changes, give me a few days for reworking the PR. But still runtime testing is needed :-p

AndrewAmmerlaan commented:
Any news on this?

efferre79 commented:
I'll probably work on it this weekend
